libs/pdi-scanner/src/protocol/parsers.rs

```diff
@@ -0,0 +1,1297 @@
+        )),
+        |(_, length_byte, unit_byte)| (length_byte, unit_byte),
+    )(input)
+}
+
+/// Parses a request to set the scan delay interval for document feed.
+///
+/// # Errors
+///
+/// Returns an error if the input does not match the expected format.
+pub fn set_scan_delay_interval_for_document_feed_request(input: &[u8]) -> IResult<&[u8], Duration> {
+    map_res(
+        packet_with_crc((tag(b"\x1bj"), le_u8)),
+        |(_, delay_interval)| {
+            if !(0x20..=0xe8).contains(&delay_interval) {
+                return Err(nom::Err::Failure(nom::error::Error::new(
+                    input,
+                    nom::error::ErrorKind::Verify,
+                )));
+            }
+
+            let delay_interval = delay_interval - 0x20;
+            Ok(Duration::from_millis(16) * delay_interval.into())
+        },
+    )(input)
+}
+
+simple_request!(
+    eject_document_to_front_of_scanner_and_hold_in_input_rollers_request,
+    b"1"
+);
+simple_request!(eject_document_to_rear_of_scanner_request, b"3");
+simple_request!(eject_document_to_front_of_scanner_request, b"4");
+simple_request!(turn_array_light_source_on_request, b"5");
+simple_request!(turn_array_light_source_off_request, b"6");
+simple_request!(eject_escrow_document_request, b"7");
+simple_request!(rescan_document_held_in_escrow_position_request, b"[");
+
+/// Parses any outgoing message of one byte or more.
+///
+/// # Errors
+///
+/// Returns an error if the input does not match the expected format.
+pub fn raw_outgoing(input: &[u8]) -> IResult<&[u8], Outgoing> {
+    if input.is_empty() {
+        Err(nom::Err::Error(nom::error::Error::new(
+            input,
+            nom::error::ErrorKind::Eof,
+        )))
+    } else {
+        Ok((&input[0..0], Outgoing::RawPacket(input.to_vec())))
+    }
+}
+
+//// Unsolicited Messages ///
+
+simple_response!(scanner_okay_event, b"#00");
+simple_response!(document_jam_event, b"#01");
+simple_response!(calibration_needed_event, b"#02");
+simple_response!(scanner_command_error_event, b"#05");
+simple_response!(read_error_event, b"#06");
+simple_response!(msd_needs_calibration_event, b"#07");
+simple_response!(msd_not_found_or_old_firmware_event, b"#08");
+simple_response!(fifo_overflow_event, b"#09");
+simple_response!(cover_open_event, b"#0C");
+simple_response!(cover_closed_event, b"#0D");
+simple_response!(command_packet_crc_error_event, b"#0E");
+simple_response!(fpga_out_of_date_event, b"#0F");
+simple_response!(calibration_ok_event, b"#10");
+simple_response!(calibration_short_calibration_document_event, b"#11");
+simple_response!(calibration_document_removed_event, b"#12");
+simple_response!(calibration_pixel_error_front_array_black, b"#13");
+simple_response!(calibration_pixel_error_front_array_white, b"#19");
+simple_response!(calibration_timeout_error, b"#1A");
+simple_response!(calibration_speed_value_error, b"#1B");
+simple_response!(calibration_speed_box_error, b"#1C");
+// TODO: fill out the rest
```

Comment:
What are the rest?

Reply:
There are more, but it's not clear to me that they will be of use to us in this driver. I'll remove this comment.